### PR TITLE
Waiv 2002,1999,1996,2202

### PR DESCRIPTION
**Pull request by oksanana**
Fixes #

### Changes

Summary of changes you made.

* Change 1
* Change 2

### Test plan

Explain how to test changes you made.

* [ ] Step 1
* [ ] Step 2

### Demo (optional)

Sreenshots (before & after) or video that show result of your work.

### CODE REVIEW

File: src/client/rewards/CampaignCardHeader/CampaignCardHeader.js [hunk 1]
```diff
@@ -2,16 +2,21 @@ import React from 'react';
...
+import { getCurrentUSDPrice } from '../rewardsHelper';
 import './CampaignCardHeader.less';
 
-const CampaignCardHeader = ({ intl, campaignData }) => {
+const CampaignCardHeader = ({ intl, campaignData, match }) => {
+  const currentUSDPrice = getCurrentUSDPrice();
   const price = get(campaignData, ['objects', '0', 'reward']);
   const isAssigned = get(campaignData, ['objects', '0', 'assigned']);
-  const rewardPriseHive = `${price ? price.toFixed(3) : 0} HIVE`;
+  const isMessages = match ? match.params.filterKey === 'messages' : null;
```
Comment (LucyHrinishyna) [inline src/client/rewards/CampaignCardHeader/CampaignCardHeader.js:14]: isMessages = match && match.params.filterKey === 'messages'

File: package.json [hunk 2]
```diff
@@ -26,11 +26,6 @@
     "heroku-prebuild": "npm i -g pm2 && pm2 install pm2-logrotate",
     "heroku-postbuild": "npm run build"
   },
-  "husky": {
```
Comment (milklim) [inline package.json:29]: why do you remove husky

File: src/client/rewards/CampaignFooter/CampaignButtons.js [hunk 3]
```diff
@@ -105,10 +129,81 @@ export default class CampaignButtons extends React.Component {
...
+            id: 'review_rejected',
+            defaultMessage: 'Review rejected',
+          }),
+        );
+      })
+      .then(() => {
+        setTimeout(() => this.props.getMessageHistory(), 8000);
+      })
+      .catch(error => {
+        console.log(error);
```
Comment (LucyHrinishyna) [inline src/client/rewards/CampaignFooter/CampaignButtons.js:161]: убери консоль, лучше выводи нотификашку

File: src/client/rewards/CampaignFooter/CampaignButtons.js [hunk 4]
```diff
@@ -174,19 +378,33 @@ export default class CampaignButtons extends React.Component {
...
-              defaultMessage: 'days left',
-            })}`}
+            {intl.formatMessage({
+              id: this.buttonsTitle.id,
+              defaultMessage: this.buttonsTitle.defaultMessage,
+            })}
+            {this.buttonsTitle.defaultMessage === 'Reserved'
+              ? `- ${daysLeft} ${intl.formatMessage({
+                  id: 'campaign_buttons_days_left',
+                  defaultMessage: 'days left',
```
Comment (LucyHrinishyna) [inline src/client/rewards/CampaignFooter/CampaignButtons.js:405]: {this.buttonsTitle.defaultMessage === 'Reserved'
              && `- ${daysLeft} ${intl.formatMessage({
                  id: 'campaign_buttons_days_left',
                  defaultMessage: 'days left',
                })} `

File: src/client/rewards/CampaignFooter/CampaignFooter.js [hunk 5]
```diff
@@ -188,17 +192,15 @@ class CampaignFooter extends React.Component {
...
-        );
-      }
+    discardPr(proposedWobj).then(() => {
+      this.toggleModal();
+      message.success(
+        this.props.intl.formatMessage({
+          id: 'discarded_successfully',
+          defaultMessage: 'Reservation released. It will be available for reservation soon.',
+        }),
+        this.props.history.push(`/rewards/active`),
```
Comment (LucyHrinishyna) [inline src/client/rewards/CampaignFooter/CampaignFooter.js:202]: для чего здесь косые кавычки `/rewards/active`

File: src/client/comments/commentsActions.js [hunk 6]
```diff
@@ -271,3 +271,34 @@ export const likeComment = (commentId, weight = 10000, vote = 'like', retryCount
...
+    payload: {
+      promise: steemConnectAPI
+        .vote(voter, currentPost.author, currentPost.permlink, weight)
+        .then(data => {
+          if (data.status === 200) {
+            return { voter, rshares: '0', percent: weight, reputation: 0 };
+          }
+          return data;
+        }),
+    },
```
Comment (milklim) [inline src/client/comments/commentsActions.js:298]: do you dispatch action without 'type'?

File: src/client/post/postActions.js [hunk 7]
```diff
@@ -95,6 +95,28 @@ export const votePost = (postId, author, permlink, weight = 10000) => (
...
+
+  if (!auth.isAuthenticated) {
+    return null;
+  }
+
+  return dispatch({
+    payload: {
+      promise: steemConnectAPI
+        .vote(voter, post.author || author, post.permlink, weight)
+        .then(res => res),
```
Comment (LucyHrinishyna) [inline src/client/post/postActions.js:115]: почему ты не указываешь тип?

File: src/client/components/PopoverMenu/PopoverMenu.js [hunk 8]
```diff
@@ -1,20 +1,24 @@
...
 
-const PopoverMenu = ({ children, onSelect, bold }) => (
+const PopoverMenu = ({ children, onSelect, bold, hide }) => (
   <ul className="PopoverMenu">
     {React.Children.map(children, child => {
       const { children: itemChildren, ...otherProps } = child.props;
+      const onItemClick = useCallback(() => {
+        onSelect(child.key);
+        hide();
+      }, [child.key, onSelect, hide]);
```
Comment (milklim) [inline src/client/components/PopoverMenu/PopoverMenu.js:13]: use only `key` as dependency

File: src/waivioApi/ApiClient.js [hunk 9]
```diff
@@ -635,6 +635,44 @@ export const getPropositions = ({
...
+    };
+    if (userName) {
+      reqData.userName = userName;
+      reqData.onlyWithMessages = false;
+    }
+    if (guideName) reqData.guideName = guideName;
+    if (!isEmpty(rewards)) reqData.rewards = rewards;
+    if (!isEmpty(status)) reqData.status = status;
+    if (!isEmpty(guideNames)) reqData.guideNames = guideNames;
+    if (!isEmpty(caseStatus)) reqData.caseStatus = caseStatus;
```
Comment (milklim) [inline src/waivioApi/ApiClient.js:665]: Do we have helper to compose request data?

File: src/common/routes.js [hunk 10]
```diff
@@ -261,16 +267,36 @@ const routes = [
...
+            exact: true,
+            component: PaymentCampaign,
+          },
           {
             path: '/rewards/receivables/@:userName',
             exact: true,
             component: PaymentCampaign,
           },
+          {
+            path: '/rewards/receivables/@:userName/:reservationPermlink',
```
Comment (milklim) [inline src/common/routes.js:281]: we can union routes: 
'/rewards/receivables/@:userName' + '/rewards/receivables/@:userName/:reservationPermlink' => 
=> '/rewards/receivables/@:userName/:reservationPermlink?'

File: src/common/routes.js [hunk 11]
```diff
@@ -261,16 +267,36 @@ const routes = [
             exact: true,
             component: PaymentCampaign,
           },
+          {
+            path: '/rewards/payables/@:userName/:reservationPermlink',
```
Comment (milklim) [inline src/common/routes.js:271]: we can union '/rewards/payables/@:userName' and '/rewards/payables/@:userName/:reservationPermlink' routes

File: src/common/routes.js [hunk 12]
```diff
@@ -225,7 +229,9 @@ const routes = [
           '/rewards/blacklist/whitelist',
           '/rewards/receivables',
           '/rewards/payables/@:userName',
+          '/rewards/payables/@:userName/:reservationPermlink',
```
Comment (milklim) [inline src/common/routes.js:232]: use one route '/rewards/payables/@:userName/:reservationPermlink?'

File: src/common/routes.js [hunk 13]
```diff
@@ -225,7 +229,9 @@ const routes = [
...
           '/rewards/receivables/@:userName',
+          '/rewards/receivables/@:userName/:reservationPermlink',
```
Comment (milklim) [inline src/common/routes.js:234]: use one route '/rewards/receivables/@:userName/:reservationPermlink?'

File: src/client/rewards/RewardsFiltersPanel/RewardsFiltersPanel.js [hunk 14]
```diff
@@ -60,30 +90,70 @@ const RewardsFiltersPanel = ({
                 defaultMessage: `Sponsors`,
               })}:`}
             </div>
-            {_.map(sponsors, sponsor =>
+            {_.map(sponsorsData, sponsor =>
```
Comment (LucyHrinishyna) [inline src/client/rewards/RewardsFiltersPanel/RewardsFiltersPanel.js:93]: не используй lodash с дефолтным импортом

File: src/client/rewards/CampaignCardHeader/CampaignCardHeader.js [hunk 15]
```diff
@@ -2,16 +2,21 @@ import React from 'react';
...
   const price = get(campaignData, ['objects', '0', 'reward']);
   const isAssigned = get(campaignData, ['objects', '0', 'assigned']);
-  const rewardPriseHive = `${price ? price.toFixed(3) : 0} HIVE`;
+  const isMessages = match && match.params.filterKey === 'messages';
+  const rewardPriseHive = `${
+    !isEmpty(price) ? price.toFixed(3) : (campaignData.reward * currentUSDPrice).toFixed(3)
+  } HIVE`;
   const rewardPriseUsd = `${campaignData.reward} USD`;
-  const rewardPrise = isAssigned ? rewardPriseHive : rewardPriseUsd;
+  const rewardPrise = isAssigned || isMessages ? rewardPriseHive : rewardPriseUsd;
```
Comment (milklim) [inline src/client/rewards/CampaignCardHeader/CampaignCardHeader.js:19]: Prise => Price

Comment (milklim) [inline src/client/rewards/CampaignCardHeader/CampaignCardHeader.js:19]: prise => price

**Comment by milklim** [comment]
resolve conflicts

**Pull request merged**